bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/FeatureTemplate.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.insteon.internal.device;
+
+import java.util.HashMap;
+import java.util.Map.Entry;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.types.Command;
+
+/**
+ * A simple class which contains the basic info needed to create a device feature.
+ * Here, all handlers are represented as strings. The actual device feature
+ * is then instantiated from the template by calling the build() function.
+ *
+ * @author Daniel Pfrommer - Initial contribution
+ * @author Rob Nielsen - Port to OpenHAB 2 insteon binding
+ */
+@NonNullByDefault
+@SuppressWarnings("null")
+public class FeatureTemplate {
+    private String m_name;
+    private String m_timeout;
+    private boolean m_isStatus;
+    private @Nullable HandlerEntry m_dispatcher = null;
+    private @Nullable HandlerEntry m_pollHandler = null;
+    private @Nullable HandlerEntry m_defaultMsgHandler = null;
+    private @Nullable HandlerEntry m_defaultCmdHandler = null;
+    private HashMap<Integer, HandlerEntry> m_messageHandlers = new HashMap<Integer, HandlerEntry>();
+    private HashMap<Class<? extends Command>, HandlerEntry> m_commandHandlers = new HashMap<Class<? extends Command>, HandlerEntry>();
```

Comment:
```suggestion
    private final String m_name;
    private final String m_timeout;
    private final boolean m_isStatus;
    private @Nullable HandlerEntry m_dispatcher = null;
    private @Nullable HandlerEntry m_pollHandler = null;
    private @Nullable HandlerEntry m_defaultMsgHandler = null;
    private @Nullable HandlerEntry m_defaultCmdHandler = null;
    private final HashMap<Integer, HandlerEntry> m_messageHandlers = new HashMap<Integer, HandlerEntry>();
    private final HashMap<Class<? extends Command>, HandlerEntry> m_commandHandlers = new HashMap<Class<? extends Command>, HandlerEntry>();
```

Reply:
This passes the code analysis checks leaving as is.